src/Communities/algorithms/pagerank.jl

```diff
@@ -1,74 +1,221 @@
+
+Helper function for compute(Pagerank, g), computes the PageRank values of the nodes in graph `g` using the classical PageRank algorithm.
 
-        # Check for convergence
-        if maximum(abs.(PR - old_PR)) < algo.tol
-            break
+# Arguments
+
+- `P::SparseMatrixCSC`: The Markov Chain associated with `g`.
+- `d::Float64`: The damping factor.
+- `tol::Float64`: The tolerance for convergence.
+- `max_iter::Int64`: The maximum number of iterations.
+
+# Returns
+
+- A vector of `Float64` where each entry represents the PageRank value of the corresponding node in the graph.
+"""
+function _classical_pagerank(
+    P::SparseMatrixCSC, 
+    d::Float64, 
+    tol::Float64, 
+    max_iter::Int64
+)::Vector{Float64}
+
+    # initialize
+    N = size(P, 1)
+
+    pr = LinearAlgebra.ones(N) / N
+
+    for _ in 1:max_iter
+        prev_pr = copy(pr)
+        # power method for principle eigenvector of modified P
+        pr = d .* P * pr .+ (1 - d) / N
+        
+        if LinearAlgebra.norm(pr - prev_pr, 1) < tol
+            return pr
         end
-        old_PR = copy(PR)
     end
 
-    # Normalize
-    PR ./= sum(PR)
+    println("Warning: max_iter reached")
+    return pr
+end
+
+"""
+    _iterative_pagerank(P::SparseMatrixCSC, d::Float64, tol::Float64, max_iter::Int64)::Vector{Float64}
+
+Helper function for compute(Pagerank, g), computes the PageRank values of the nodes in graph `g` using the iterative regularization PageRank algorithm.
 
-    return PR
+# Arguments
+
+- `P::SparseMatrixCSC`: The Markov Chain associated with `g`.
+- `d::Float64`: The initial damping factor, from which the regularization factor is computed. 0 < d < .5
+- `tol::Float64`: The tolerance for convergence.
+- `max_iter::Int64`: The maximum number of iterations.
+
+# Returns
+
+- A vector of `Float64` where each entry represents the PageRank value of the corresponding node in the graph.
+"""
+function _iterative_regularization_pagerank(
+    P::SparseMatrixCSC, 
+    d::Float64, 
+    tol::Float64, 
+    max_iter::Int64
+)::Vector{Float64}
+
+    # initialize
+    N = size(P, 1)
+
+    pr = LinearAlgebra.ones(N) / N
+    # compute p for regularization factor
+    p = log2(1 / (1 - d))
+
+    # iterate until tolerance is reached or max_iter is reached
+    for k in 1:max_iter
+        # regularization factor
+        α = 1 - 1 / ((k+2) ^ p) 
+
+        prev_pr = copy(pr)
+        # power method for principle eigenvector of modified P
+        pr = α .* P * pr .+ (1 - α) / N
+        
+        if LinearAlgebra.norm(pr - prev_pr, 1) < tol
+            println("Converged in $k iterations")
+            return pr
+        end
+    end
+    println("Warning: max_iter reached")
+    
+    return pr
 end
+
+"""
+    _adaptive_pagerank(P::SparseMatrixCSC, d::Float64, tol::Float64, max_iter::Int64)::Vector{Float64}
+
+Helper function for compute(Pagerank, g), computes the PageRank values of the nodes in graph `g` using the adaptive regularization PageRank algorithm.
```

Comment:
Can we keep the number of characters used on a single line under 90 characters? I know this isn't the case everywhere, but we will want to do this before we reach our first real version of the package.